### DISCUSSION TITLE
Notifications and Tray Icon

### DISCUSSION THREAD

**Discussion by Steve0212a** [dotnet/maui#13937] [Q&A]
I am in the process of writing an app for a company that will interact with their UPSes (think powerchute or powerpanel).  We want it to run on Windows and Mac (and later some flavors of Linux).  I have done a lot of WPF development and Maui seems very similar.  

I do have some questions though:

1. Can a Maui app have a system tray icon on Windows and Mac?  I think the answer is no for Mac, but wanted to check
2. A big part of a UPS app is the notifications for the user (i.e. power is out, your computer will shut down in x minutes).  On Windows and Mac, can Maui display notifications to the user if the app is not running like on mobile?
3. Is there any way to support Linux now?  I have seen some posts about integrating Blazor with Maui and that could work.
4. Will there be Linux support in the future?

I did see this post, https://github.com/dotnet/maui/discussions/130, but it did not really provide much info.

I could probably get by without the tray icon if I could get notifications working while the app is not running.  However, if I need the app running, it would need to be closed to the tray area.

Thanks.

**Comment by Steve0212a** [comment]
A friend pointed me to this in case it helps.  this app does have a tray icon and notifications in Mac

https://github.com/dotnet/maui-samples/tree/main/6.0/Apps/WeatherTwentyOne

**Comment by MV10** [comment]
On Windows, the app has to be running to show notifications (aka "Toasts") and it also has to be a user-desktop app (not a service). I wrote some about this on my [blog](https://mcguirev10.com/2023/05/14/ups-monitor-battery-backup-event-notifications.html), ironically in a simple little UPS monitoring app (not WPF or MAUI) and my code is in my repo.

You can see where notifications are shown [here](https://github.com/MV10/UPSMonitor/blob/master/UPSMonitor/MessageServer.cs#L45) in my repo (which the system tray portion receives over a named pipe from the background service).

I'm also hoping MS gets with the program and adds X11 support to MAUI. Hard to believe they support a nearly useless niche like Tizen but not X11 (yes, I know it's a side-effect of the Xamarin roots).
